src/main/java/org/zalando/nakadi/domain/BatchItem.java

```diff
@@ -58,4 +59,8 @@ public void updateStatusAndDetail(final EventPublishingStatus publishingStatus,
         response.setPublishingStatus(publishingStatus);
         response.setDetail(detail);
     }
+
+    public int getEventSize() {
+        return event.toString().getBytes(StandardCharsets.UTF_8).length;
```

Comment:
@antban @rcillo We've discussed something similar on Jira ticket 511. What I've been doing here is basically telling users "You can't send any event larger than 1,000,000 bytes". Pro: it's easy for producers to know how large their event can be; Con: we have to take into account that enrichment and metadata will increase the size of the event, so we have to configure Kafka to allow for events that are larger than 1,000,000 bytes.

The solution @antban mentions is different, I think: "We will not store events larger than 1,000,000 bytes". This means that the actual max size for a producer will be a bit lower, depending on the enrichment strategies selected. Pro: We know exactly how large the largest events will be in Kafka, and we can configure its maximum size very easily; Con: it's hard for producers to tell in advance whether an event will be accepted or not (if it's close to the limit).

Whatever solution we choose, I agree with @antban that we should improve BatchItem to reduce the number of toString() operations, but this can perhaps be done with another ticket.